pilot/pkg/model/push_context.go

```diff
@@ -1587,7 +1587,7 @@ func (ps *PushContext) initSidecarScopes(env *Environment) error {
 	ps.sidecarIndex.sidecarsByNamespace = make(map[string][]*SidecarScope, sidecarNum)
 	for i, sidecarConfig := range sidecarConfigs {
 		ps.sidecarIndex.sidecarsByNamespace[sidecarConfig.Namespace] = append(ps.sidecarIndex.sidecarsByNamespace[sidecarConfig.Namespace],
-			ConvertToSidecarScope(ps, &sidecarConfig, sidecarConfig.Namespace))
```

Comment:
This was wrong, SidecarScope.Sidecar record the  address of sidecarConfig

Can you add a test

Reply:
OK, I'll add ut.

Reply:
Sorry, after another look, the SidecarScope.Sidecar reference sidecarConfig.Spec. No problem. 

This can be shown: the test pass with/without this patch.